src/js/Parser.js

```diff
@@ -0,0 +1,92 @@
+import { Station } from './Station';
+import { SubwayLine } from './SubwayLine';
+import { Subway } from './Subway';
+
+export function parseToSubway(data) {
+    let currentData = data;
+
+    currentData = handleToArray(currentData);
+    currentData = handleToSubway(currentData);
+
+
+    return createSubway(currentData);
+} 
+
+function handleToArray(data) {
+    let array = data;
+    array = array.split('\n');
+    array = Array.from(array);
+
+    array = array.map(name => {
+        return name.trim();
+    });
+
+    return array;
+}
+
+function handleToSubway(data) {
+    let stationNames = [], names = [], lineNames = [];
+    let lineName = null;
+
+    data.forEach(name => {
+        if (!name || data.lastIndexOf(name) === data.length - 1) {
```

Comment:
you introduce additonal algorithmic complexity using lastIndexOf. The first one you can use simple for and have index and determine index of current element without lastIndexOf. The second one forEach() has the second parameter index.
Also you call it twice!